pkg/cluster/cluster.go

```diff
@@ -63,12 +66,22 @@ func randString(n int) string {
 const (
 	validNameRegexpStr = "^[a-zA-Z0-9-]{1,50}$"
 	weaveNet           = "https://github.com/weaveworks/weave/releases/download/v2.0.5/weave-daemonset-k8s-1.7.yaml"
+	flannelNet         = "https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml"
+	calicoRBAC         = "https://docs.projectcalico.org/v3.1/getting-started/kubernetes/installation/hosted/rbac-kdd.yaml"
+	calicoNet          = "https://docs.projectcalico.org/v3.1/getting-started/kubernetes/installation/hosted/kubernetes-datastore/calico-networking/1.7/calico.yaml"
 
 	// Avoid token passing between master and worker nodes by using
 	// a hard-coded token
 	kubeadmToken = "aaaaaa.bbbbbbbbbbbbbbbb"
 )
 
+var cniFiles = map[string][]string{
+	"base":    {"bridge", "cnitool", "dhcp", "host-device", "host-local", "ipvlan", "loopback", "macvlan", "noop", "portmap", "ptp", "sample", "tuning", "vlan"},
```

Comment:
I don't think it's necessary to copy `cnitool` and `noop` here.
[CNI plugins v0.7.1](https://github.com/containernetworking/plugins/releases/tag/v0.7.1) has only the following plugins: 
`bridge`, `dhcp`, `host-device`, `host-local`, `ipvlan`, `loopback`, `macvlan`, `portmap`, `ptp`, `sample`, `tuning`, `vlan`, and `flannel`.

As a result, if you only installed those 13 CNI plugins (12 base + 1 flannel), kube-spawn will fail to start.
Actually I had no problem with running kube-spawn, even without installing `noop` and `cnitool`.
So can you please remove `noop` and `cnitool` from the base map?

Besides, you would also pre-install calico plugins, because these are not included in CNI plugins. Though that's not a big deal. Calico users should be able to deal with the issue.